test/functional/error_tests.js

```diff
@@ -1,440 +1,139 @@
 'use strict';
-var test = require('./shared').assert;
-var setupDatabase = require('./shared').setupDatabase;
+const setupDatabase = require('./shared').setupDatabase;
+const chai = require('chai');
+const expect = chai.expect;
+const sinonChai = require('sinon-chai');
+chai.use(sinonChai);
 
-describe.skip('Errors', function() {
+describe('Errors', function() {
   before(function() {
     return setupDatabase(this.configuration);
   });
 
-  it('shouldFailInsertDueToUniqueIndex', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
-      client.connect(function(err, client) {
-        var db = client.db(configuration.db);
-        var collection = db.collection('test_failing_insert_due_to_unique_index');
-        collection.ensureIndex([['a', 1]], { unique: true, w: 1 }, function(err) {
-          test.equal(null, err);
-
-          collection.insert({ a: 2 }, { w: 1 }, function(err) {
-            test.ok(err == null);
-
-            collection.insert({ a: 2 }, { w: 1 }, function(err) {
-              test.ok(err.code != null);
-              test.ok(err != null);
-              client.close();
-              done();
+  it('should fail insert due to unique index', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
+    client.connect((err, client) => {
+      const db = client.db(configuration.db);
+      const collection = db.collection('test_failing_insert_due_to_unique_index');
+      collection.createIndexes(
+        [
+          {
+            name: 'test_failing_insert_due_to_unique_index',
+            key: { a: 1 },
+            unique: true
+          }
+        ],
+        { w: 1 },
+        err => {
+          expect(err).to.not.exist;
+
+          collection.insertOne({ a: 2 }, { w: 1 }, err => {
+            expect(err).to.not.exist;
+
+            collection.insertOne({ a: 2 }, { w: 1 }, err => {
+              expect(err).to.exist;
```

Comment:
Since we're checking the code and message below, we can delete `expect(err).to.exist`.